contracts/tips/src/msg.rs

```diff
@@ -0,0 +1,282 @@
+use crate::error::ContractError;
+use crate::state::StateServiceFee;
+use cosmwasm_std::{Addr, Coin, Uint128, Uint64};
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+/// Fees required to execute [`ExecuteMsg::SendTip`].
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum ServiceFee {
+    /// Represents a fixed fee that is deducted from the tip.
```

Comment:
I see that both the `Fixed` and `Percentage` type of fees state the following: 
> [...] fee that is deducted from the tip.

From a user perspective, it looks like if the user wants to send 1 DSM and the fee is 0.01 DSM: 
- the sender will end up paying 1 DSM 
- the receiver will receive 0.99 DSM (1 DSM - 0.01 DSM)

Although this is works, I think that instead the fee amount should be payed by the sender, and not subtracted from the tip amount. So, if a sender wants to send a 1 DSM tip and the fee is 0.01 DSM the following should happen: 
- the sender pays 1.01 DSM 
- the receiver pays 0.99 DSM

This will also make it easier to be displayed in a UI and will cause less confusion from the sender's perspective. 

If this is already the case of how things work, I think this comment should be edited into something like
> [...] fee that the sender will have to pay.

Reply:
In the example of the desired behaviour you mean
* the sender pays 1.01 DSM
* the receiver receives 1 DSM  

If so the desired behaviour that you are describing is already implemented for the `Fixed` but not for `Percentage`.
I can extend the `SendTip` message to have a `amount` field that represents the amount that the sender want tip and calculate the percentage from this `amount` field, then if `calculated fee` + `amount` is equal to the coins received in the `MessageInfo::amount` we consider the tip valid.  

cc @bragaz